Post: Lifehacking 101: I'm bad at to-do lists, now what? (#216)
mauroservienti/milestone.github.io: _posts/bad-at-to-do-lists-now-what.md

Running on zero-notifications is a relief; with no pressure coming from badges or any other sort of distracting and pressure-designed crap, my day is already doing great.

However, as already said in the above-linked article, the day in Particular Software is a bit particular (no pun intended). Time zones, multiple task forces, wearing more than one hat ranging from "product management" to support, all participate in the risk of making the day a bit of a mess. 

## Please welcome context switching 

One of the side effects of wearing multiple hats throughout the day is the rise of context switching. Let's do some extreme math to put the issue in context; if we were to have only 5-minute long tasks and an 8-hour workday, the hypothetical result is 96 completed tasks: 12 tasks every hour, times 8 hours. Now, factor in multiple hats, and again for the sake of the sample, let's imagine that each assignment requires a different hat and that to change it (context switch), it takes our brain about 5 minutes. In this scenario, we'll be able to complete 50% of the 96 tasks each day — a colossal dropoff.

> One could argue a zero-context switching approach would be an ideal world. Probably. I guess it depends on the person and the task. There are things that I enjoy a lot, like writing code or writing in general, that bores me very quickly; after a couple of hours in the best case, my mind starts digressing, signaling it's time for a change.

Author: @kbaley what about something like:

```suggestion
One could argue a zero-context switching approach would be an ideal world. Probably. I guess it depends on the person and the task. For example, when it comes to me, there are things, like writing or programming, that I enjoy a lot. Simultaneously, the excitement starts to sunset, usually in a couple of hours. My mind starts digressing, signaling it's time for a change.
```

I've been running on zero-notifications for the last five or more years. Zero-notifications means that my computer, or any other device, have notifications turned off for everything except the following:

- The calendar on the computer and phone
- Phone calls and messages, including text messages, Signal, and WhatsApp. All groups, without exceptions, get muted on day one.
Nothing else. And when it comes to phone calls and messages, my iPhone turns on "Do Not Disturb" automatically from 6 PM to  8 AM, in which case only 6 or 7 numbers can dial in. That is it.

Running on zero-notifications is a relief; with no pressure coming from badges or any other sort of distracting and pressure-designed crap, my day is already doing great.

However, as already said in the above-linked article, the day in Particular Software is a bit particular (no pun intended). Time zones, multiple task forces, wearing more than one hat ranging from "product management" to support, all participate in the risk of making the day a bit of a mess. 

## Please welcome context switching 

> One of the side effects of wearing multiple hats throughout the day is the rise of context switching. Let's do some extreme math to put the issue in context; if we were to have only 5-minute long tasks and an 8-hour workday, the hypothetical result is 96 completed tasks: 12 tasks every hour, times 8 hours. Now, factor in multiple hats, and again for the sake of the sample, let's imagine that each assignment requires a different hat and that to change it (context switch), it takes our brain about 5 minutes. In such a scenario, we'll be able to complete 50% of the 96 tasks a day—a colossal cutoff.

Reviewer: ```suggestion
One of the side effects of wearing multiple hats throughout the day is the rise of context switching. Let's do some extreme math to put the issue in context; if we were to have only 5-minute long tasks and an 8-hour workday, the hypothetical result is 96 completed tasks: 12 tasks every hour, times 8 hours. Now, factor in multiple hats, and again for the sake of the sample, let's imagine that each assignment requires a different hat and that to change it (context switch), it takes our brain about 5 minutes. In this scenario, we'll be able to complete 50% of the 96 tasks each day — a colossal dropoff.
```

> One could argue a zero-context switching approach would be an ideal world. Probably. I guess it depends on the person and the task. There are things that I enjoy a lot, like writing code or writing in general, that bores me very quickly; after a couple of hours in the best case, my mind starts digressing, signaling it's time for a change.

Reviewer: > There are things that I enjoy a lot, like writing code or writing in general, that bores me very quickly

This looks contradictory. Do you mean you enjoy writing code but that writing in general bores you?

Author: I mean enjoy both of them, but both of them bore me very quickly. So, I wanna write code but not for 8 hours a day

Reviewer: In that case:

> There are things that I enjoy a lot but that bore me very quickly, like writing code or writing in general.

> > If there is an unsolved problem, like a bug or a not yet wholly identified solution, that's another story.

Reviewer: ```suggestion
> If there is an unsolved problem, like a bug or a not-yet wholly-identified solution, that's another story.
```

> I also enjoy many diverse activities ranging from solving technical problems to "product management" to more business-related activities,

Reviewer: ```suggestion
I also enjoy many diverse activities ranging from solving technical problems to "product management" to more business-related activities.
```

Combine this with the next paragraph

However, if I follow my brain's hint and switch to another activity, the effort required is extreme. In retrospect, I can state that the first 15 to 20 minutes of the next task are "wasted" cycles in the attempt to clear up my mind and get the necessary concentration to attack the new problem.

> > On a side note, that is the reason why I love "commuting" on foot. My office is precisely a mile from home. In the best case, it's an 8-minute walk (I'm in a hurry sometimes) or a 15-minute relaxing one. Either one is the perfect decompression time. By the time I get home, my brain is ready for the 3-year old little Dracula ;-)

Reviewer: ```suggestion
> On a side note, that is the reason why I love "commuting" on foot. My office is precisely a mile from home. That's a 15-minute leisurely walk or an 8-minute brisk walk. Either one is a good decompression time. By the time I get home, my brain is ready for the 3-year old little Dracula ;-)
```

## Fillers

> Since a couple of months, so it's probably too early for some conclusions, I started injecting filler activities between daily tasks. One of the things I enjoy is reading.

Reviewer: ```suggestion
For a couple of months, so it's probably too early for some conclusions, I started injecting filler activities between daily tasks. One of the things I enjoy is reading.
```

Consider expanding on what you mean here by "filler" activity. For example:

```suggestion
For a couple of months, so it's probably too early for some conclusions, I started injecting filler activities between daily tasks. These are leisure activities that I enjoy doing and that I can pick up and put down in short bursts. That is, they don't require a lot of mental activity to start or to stop.
```

> Reading turned out to be an excellent filler activity. Building LEGO is another satisfactory one.

Reviewer: If you accept my second suggestion above, combine this sentence with the previous paragraph:

```suggestion
For me, reading is an excellent filler activity. Building LEGO is another satisfactory one.
```

I extensively use [Pocket](https://getpocket.com) as the tool to keep track of all the articles I want to read. I carefully tag articles that I continuously store. I tend to read about everything except technology; if you could look at my tags, very few items are labeled as `technology` or `architecture`. My reading interests are mostly politics, economics, mindfulness, psychology, and art.

Whenever I need to switch to a different task, I merely open Pocket and select a couple or more articles to read, not more than 10 minutes or even less. 

> I keep an eye on the topics; I carefully select topics that are very distant from what I was doing right before and from what I'm going to do next. Results are dual:

Reviewer: ```suggestion
I keep an eye on the topics; I carefully select topics that are very distant from what I was doing right before and from what I'm going to do next. The result is two-fold:
```

- I feed my soul (typical Italian idiom that has nothing to do with religion in this case),
- I clear my brain from the leftovers of the previous task and get it ready for the next.

## Conclusion

> Context switching is a problem, that's no doubt, mainly if it plays a primary role in our days. On the one hand, it's essential to reduce context switching to the minimum required; on the other hand, we can benefit from it, at least I did. It comes with a positive effect for people like me that need to have variety throughout their day.

Reviewer: ```suggestion
There's no doubt context switching is a problem and that it plays a key role in our days. On the one hand, it's essential to reduce context switching to the minimum required; on the other hand, we can benefit from it, at least I did. It comes with a positive effect for people like me that need to have variety throughout their day.
```
